[joltphysics] Build failure

Package: joltphysics:x64-linux -> 3.0.1#1

**Host Environment**

- Host: x64-linux
- Compiler: GNU 13.2.0
-    vcpkg-tool version: 2023-10-18-27de5b69dac4b6fe8259d283cd4011e6d20a84ce
    vcpkg-scripts version: 830f86fb3 2023-10-20 (24 hours ago)

**To Reproduce**

`vcpkg install `
**Failure logs**

```
-- Using cached jrouwe-JoltPhysics-v3.0.1.tar.gz.
-- Cleaning sources at /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/dennis/src/vcpkg/downloads/jrouwe-JoltPhysics-v3.0.1.tar.gz
-- Applying patch fix-export.diff
-- Using source at /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean
-- Found external ninja('1.11.1.git.kitware.jobserver-1').
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/cmake --build . --config Debug --target install -- -v -j9
    Working Directory: /home/dennis/src/vcpkg/buildtrees/joltphysics/x64-linux-dbg
    See logs for more information:
      /home/dennis/src/vcpkg/buildtrees/joltphysics/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  /home/dennis/src/WickedEngine/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  /home/dennis/src/WickedEngine/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/joltphysics/portfile.cmake:35 (vcpkg_cmake_install)
  scripts/ports.cmake:168 (include)



```
<details><summary>/home/dennis/src/vcpkg/buildtrees/joltphysics/install-x64-linux-dbg-out.log</summary>

```
Change Dir: '/home/dennis/src/vcpkg/buildtrees/joltphysics/x64-linux-dbg'

Run Build Command(s): /home/dennis/.local/bin/ninja -v -v -j9 install
[1/134] /usr/bin/c++ -DJPH_DEBUG_RENDERER -DJPH_PROFILE_ENABLED -DJPH_USE_AVX -DJPH_USE_AVX2 -DJPH_USE_F16C -DJPH_USE_FMADD -DJPH_USE_LZCNT -DJPH_USE_SSE4_1 -DJPH_USE_SSE4_2 -DJPH_USE_TZCNT -D_DEBUG -I/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/.. -fPIC -I. -g -Wno-comment -Wno-stringop-overflow -ffp-contract=off -g -std=c++17 -mavx2 -mbmi -mpopcnt -mlzcnt -mf16c -mfma -Winvalid-pch -x c++-header -include /home/dennis/src/vcpkg/buildtrees/joltphysics/x64-linux-dbg/CMakeFiles/Jolt.dir/cmake_pch.hxx -MD -MT CMakeFiles/Jolt.dir/cmake_pch.hxx.gch -MF CMakeFiles/Jolt.dir/cmake_pch.hxx.gch.d -o CMakeFiles/Jolt.dir/cmake_pch.hxx.gch -c /home/dennis/src/vcpkg/buildtrees/joltphysics/x64-linux-dbg/CMakeFiles/Jolt.dir/cmake_pch.hxx.cxx
FAILED: CMakeFiles/Jolt.dir/cmake_pch.hxx.gch 
/usr/bin/c++ -DJPH_DEBUG_RENDERER -DJPH_PROFILE_ENABLED -DJPH_USE_AVX -DJPH_USE_AVX2 -DJPH_USE_F16C -DJPH_USE_FMADD -DJPH_USE_LZCNT -DJPH_USE_SSE4_1 -DJPH_USE_SSE4_2 -DJPH_USE_TZCNT -D_DEBUG -I/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/.. -fPIC -I. -g -Wno-comment -Wno-stringop-overflow -ffp-contract=off -g -std=c++17 -mavx2 -mbmi -mpopcnt -mlzcnt -mf16c -mfma -Winvalid-pch -x c++-header -include /home/dennis/src/vcpkg/buildtrees/joltphysics/x64-linux-dbg/CMakeFiles/Jolt.dir/cmake_pch.hxx -MD -MT CMakeFiles/Jolt.dir/cmake_pch.hxx.gch -MF CMakeFiles/Jolt.dir/cmake_pch.hxx.gch.d -o CMakeFiles/Jolt.dir/cmake_pch.hxx.gch -c /home/dennis/src/vcpkg/buildtrees/joltphysics/x64-linux-dbg/CMakeFiles/Jolt.dir/cmake_pch.hxx.cxx
In file included from /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/..//Jolt/Jolt.h:8,
                 from /home/dennis/src/vcpkg/buildtrees/joltphysics/x64-linux-dbg/CMakeFiles/Jolt.dir/cmake_pch.hxx:5,
                 from <command-line>:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Core/Core.h:310:15: error: ‘uint8_t’ does not name a type
  310 | using uint8 = uint8_t;
      |               ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Core/Core.h:272:1: note: ‘uint8_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
  271 |         #include <immintrin.h>
  +++ |+#include <cstdint>
  272 | #elif defined(JPH_USE_NEON)
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Core/Core.h:311:16: error: ‘uint16_t’ does not name a type
  311 | using uint16 = uint16_t;
      |                ^~~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Core/Core.h:311:16: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Core/Core.h:312:16: error: ‘uint32_t’ does not name a type
...
Skipped 353 lines
...
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec3.inl: In member function ‘float JPH::Vec3::ReduceMax() const’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec3.inl:807:76: error: no matching function for call to ‘JPH::Vec3::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_UNUSED, JPH::SWIZZLE_Z>() const’
  807 |         Vec3 v = sMax(mValue, Swizzle<SWIZZLE_Y, SWIZZLE_UNUSED, SWIZZLE_Z>());
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec3.h:195:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error> > JPH::Vec3 JPH::Vec3::Swizzle() const’
  195 |         JPH_INLINE Vec3                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec3.h:195:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec3.inl:808:73: error: no matching function for call to ‘JPH::Vec3::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_UNUSED, JPH::SWIZZLE_UNUSED>()’
  808 |         v = sMax(v, v.Swizzle<SWIZZLE_Z, SWIZZLE_UNUSED, SWIZZLE_UNUSED>());
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec3.h:195:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error> > JPH::Vec3 JPH::Vec3::Swizzle() const’
  195 |         JPH_INLINE Vec3                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec3.h:195:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl: At global scope:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:46:10: error: ‘uint32’ has not been declared
   46 | template<uint32 SwizzleX, uint32 SwizzleY, uint32 SwizzleZ, uint32 SwizzleW>
      |          ^~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:46:27: error: ‘uint32’ has not been declared
   46 | template<uint32 SwizzleX, uint32 SwizzleY, uint32 SwizzleZ, uint32 SwizzleW>
      |                           ^~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:46:44: error: ‘uint32’ has not been declared
   46 | template<uint32 SwizzleX, uint32 SwizzleY, uint32 SwizzleZ, uint32 SwizzleW>
      |                                            ^~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:46:61: error: ‘uint32’ has not been declared
   46 | template<uint32 SwizzleX, uint32 SwizzleY, uint32 SwizzleZ, uint32 SwizzleW>
      |                                                             ^~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl: In member function ‘JPH::Vec4 JPH::Vec4::Swizzle() const’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:49:23: error: ‘SwizzleX’ was not declared in this scope; did you mean ‘Swizzle’?
   49 |         static_assert(SwizzleX <= 3, "SwizzleX template parameter out of range");
      |                       ^~~~~~~~
      |                       Swizzle
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:50:23: error: ‘SwizzleY’ was not declared in this scope; did you mean ‘Swizzle’?
   50 |         static_assert(SwizzleY <= 3, "SwizzleY template parameter out of range");
      |                       ^~~~~~~~
      |                       Swizzle
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:51:23: error: ‘SwizzleZ’ was not declared in this scope; did you mean ‘Swizzle’?
   51 |         static_assert(SwizzleZ <= 3, "SwizzleZ template parameter out of range");
      |                       ^~~~~~~~
      |                       Swizzle
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:52:23: error: ‘SwizzleW’ was not declared in this scope; did you mean ‘Swizzle’?
   52 |         static_assert(SwizzleW <= 3, "SwizzleW template parameter out of range");
      |                       ^~~~~~~~
      |                       Swizzle
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl: In static member function ‘static void JPH::Vec4::sSort4(JPH::Vec4&, JPH::UVec4&)’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:304:78: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_W, JPH::SWIZZLE_X, JPH::SWIZZLE_Y>()’
  304 |         Vec4 v1 = ioValue.Swizzle<SWIZZLE_Z, SWIZZLE_W, SWIZZLE_X, SWIZZLE_Y>();
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:305:79: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_W, JPH::SWIZZLE_X, JPH::SWIZZLE_Y>()’
  305 |         UVec4 i1 = ioIndex.Swizzle<SWIZZLE_Z, SWIZZLE_W, SWIZZLE_X, SWIZZLE_Y>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:306:90: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_W, JPH::SWIZZLE_Z, JPH::SWIZZLE_W>()’
  306 |         UVec4 c1 = sLess(ioValue, v1).Swizzle<SWIZZLE_Z, SWIZZLE_W, SWIZZLE_Z, SWIZZLE_W>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:311:78: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_X, JPH::SWIZZLE_W, JPH::SWIZZLE_Z>()’
  311 |         Vec4 v2 = ioValue.Swizzle<SWIZZLE_Y, SWIZZLE_X, SWIZZLE_W, SWIZZLE_Z>();
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:312:79: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_X, JPH::SWIZZLE_W, JPH::SWIZZLE_Z>()’
  312 |         UVec4 i2 = ioIndex.Swizzle<SWIZZLE_Y, SWIZZLE_X, SWIZZLE_W, SWIZZLE_Z>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:313:90: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_Y, JPH::SWIZZLE_W, JPH::SWIZZLE_W>()’
  313 |         UVec4 c2 = sLess(ioValue, v2).Swizzle<SWIZZLE_Y, SWIZZLE_Y, SWIZZLE_W, SWIZZLE_W>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:318:78: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_X, JPH::SWIZZLE_Z, JPH::SWIZZLE_Y, JPH::SWIZZLE_W>()’
  318 |         Vec4 v3 = ioValue.Swizzle<SWIZZLE_X, SWIZZLE_Z, SWIZZLE_Y, SWIZZLE_W>();
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:319:79: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_X, JPH::SWIZZLE_Z, JPH::SWIZZLE_Y, JPH::SWIZZLE_W>()’
  319 |         UVec4 i3 = ioIndex.Swizzle<SWIZZLE_X, SWIZZLE_Z, SWIZZLE_Y, SWIZZLE_W>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:320:90: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_X, JPH::SWIZZLE_Z, JPH::SWIZZLE_Z, JPH::SWIZZLE_W>()’
  320 |         UVec4 c3 = sLess(ioValue, v3).Swizzle<SWIZZLE_X, SWIZZLE_Z, SWIZZLE_Z, SWIZZLE_W>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl: In static member function ‘static void JPH::Vec4::sSort4Reverse(JPH::Vec4&, JPH::UVec4&)’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:328:78: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_W, JPH::SWIZZLE_X, JPH::SWIZZLE_Y>()’
  328 |         Vec4 v1 = ioValue.Swizzle<SWIZZLE_Z, SWIZZLE_W, SWIZZLE_X, SWIZZLE_Y>();
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:329:79: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_W, JPH::SWIZZLE_X, JPH::SWIZZLE_Y>()’
  329 |         UVec4 i1 = ioIndex.Swizzle<SWIZZLE_Z, SWIZZLE_W, SWIZZLE_X, SWIZZLE_Y>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:330:93: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_W, JPH::SWIZZLE_Z, JPH::SWIZZLE_W>()’
  330 |         UVec4 c1 = sGreater(ioValue, v1).Swizzle<SWIZZLE_Z, SWIZZLE_W, SWIZZLE_Z, SWIZZLE_W>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:335:78: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_X, JPH::SWIZZLE_W, JPH::SWIZZLE_Z>()’
  335 |         Vec4 v2 = ioValue.Swizzle<SWIZZLE_Y, SWIZZLE_X, SWIZZLE_W, SWIZZLE_Z>();
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:336:79: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_X, JPH::SWIZZLE_W, JPH::SWIZZLE_Z>()’
  336 |         UVec4 i2 = ioIndex.Swizzle<SWIZZLE_Y, SWIZZLE_X, SWIZZLE_W, SWIZZLE_Z>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:337:93: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_Y, JPH::SWIZZLE_W, JPH::SWIZZLE_W>()’
  337 |         UVec4 c2 = sGreater(ioValue, v2).Swizzle<SWIZZLE_Y, SWIZZLE_Y, SWIZZLE_W, SWIZZLE_W>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:342:78: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_X, JPH::SWIZZLE_Z, JPH::SWIZZLE_Y, JPH::SWIZZLE_W>()’
  342 |         Vec4 v3 = ioValue.Swizzle<SWIZZLE_X, SWIZZLE_Z, SWIZZLE_Y, SWIZZLE_W>();
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:343:79: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_X, JPH::SWIZZLE_Z, JPH::SWIZZLE_Y, JPH::SWIZZLE_W>()’
  343 |         UVec4 i3 = ioIndex.Swizzle<SWIZZLE_X, SWIZZLE_Z, SWIZZLE_Y, SWIZZLE_W>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:344:93: error: no matching function for call to ‘JPH::UVec4::Swizzle<JPH::SWIZZLE_X, JPH::SWIZZLE_Z, JPH::SWIZZLE_Z, JPH::SWIZZLE_W>()’
  344 |         UVec4 c3 = sGreater(ioValue, v3).Swizzle<SWIZZLE_X, SWIZZLE_Z, SWIZZLE_Z, SWIZZLE_W>();
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::UVec4 JPH::UVec4::Swizzle() const’
   39 |         JPH_INLINE UVec4                        Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:39:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl: In member function ‘float JPH::Vec4::ReduceMin() const’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:760:92: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_UNUSED, JPH::SWIZZLE_W, JPH::SWIZZLE_UNUSED>() const’
  760 |         Vec4 v = sMin(mValue, Swizzle<SWIZZLE_Y, SWIZZLE_UNUSED, SWIZZLE_W, SWIZZLE_UNUSED>());
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:761:89: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_UNUSED, JPH::SWIZZLE_UNUSED, JPH::SWIZZLE_UNUSED>()’
  761 |         v = sMin(v, v.Swizzle<SWIZZLE_Z, SWIZZLE_UNUSED, SWIZZLE_UNUSED, SWIZZLE_UNUSED>());
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl: In member function ‘float JPH::Vec4::ReduceMax() const’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:767:92: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_UNUSED, JPH::SWIZZLE_W, JPH::SWIZZLE_UNUSED>() const’
  767 |         Vec4 v = sMax(mValue, Swizzle<SWIZZLE_Y, SWIZZLE_UNUSED, SWIZZLE_W, SWIZZLE_UNUSED>());
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Vec4.inl:768:89: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_UNUSED, JPH::SWIZZLE_UNUSED, JPH::SWIZZLE_UNUSED>()’
  768 |         v = sMax(v, v.Swizzle<SWIZZLE_Z, SWIZZLE_UNUSED, SWIZZLE_UNUSED, SWIZZLE_UNUSED>());
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
In file included from /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl:9,
                 from /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Mat44.h:236,
                 from /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/..//Jolt/Jolt.h:15:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Quat.h: In member function ‘JPH::Quat JPH::Quat::Conjugated() const’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Quat.h:168:237: error: no matching function for call to ‘JPH::UVec4::UVec4(unsigned int, unsigned int, unsigned int, int)’
  168 |         JPH_INLINE Quat                         Conjugated() const                                                                                              { return Quat(Vec4::sXor(mValue, UVec4(0x80000000, 0x80000000, 0x80000000, 0).ReinterpretAsFloat())); }
      |                                                                                                                                                                                                                                             ^
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:28:57: note: candidate: ‘JPH::UVec4::UVec4(Type)’
   28 |         JPH_INLINE                                      UVec4(Type inRHS) : mValue(inRHS)                                       { }
      |                                                         ^~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:28:57: note:   candidate expects 1 argument, 4 provided
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:27:65: note: candidate: ‘constexpr JPH::UVec4::UVec4(const JPH::UVec4&)’
   27 |                                                                 UVec4(const UVec4 &inRHS) = default;
      |                                                                 ^~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:27:65: note:   candidate expects 1 argument, 4 provided
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:26:65: note: candidate: ‘JPH::UVec4::UVec4()’
   26 |                                                                 UVec4() = default; ///< Intentionally not initialized for performance reasons
      |                                                                 ^~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:26:65: note:   candidate expects 0 arguments, 4 provided
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Quat.h: In member function ‘JPH::Quat JPH::Quat::GetPerpendicular() const’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Quat.h:177:246: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_X, JPH::SWIZZLE_W, JPH::SWIZZLE_Z>() const’
  177 |         JPH_INLINE Quat                         GetPerpendicular() const                                                                                { return Quat(Vec4(1, -1, 1, -1) * mValue.Swizzle<SWIZZLE_Y, SWIZZLE_X, SWIZZLE_W, SWIZZLE_Z>()); }
      |                                                                                                                                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
In file included from /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Quat.h:245:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Quat.inl: In static member function ‘static JPH::Quat JPH::Quat::sRotation(JPH::Vec3Arg, float)’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Quat.inl:80:78: error: no matching function for call to ‘JPH::UVec4::UVec4(int, int, int, unsigned int)’
   80 |     return Quat(Vec4::sSelect(Vec4(inAxis) * s, c, UVec4(0, 0, 0, 0xffffffffU)));
      |                                                                              ^
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:28:57: note: candidate: ‘JPH::UVec4::UVec4(Type)’
   28 |         JPH_INLINE                                      UVec4(Type inRHS) : mValue(inRHS)                                       { }
      |                                                         ^~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:28:57: note:   candidate expects 1 argument, 4 provided
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:27:65: note: candidate: ‘constexpr JPH::UVec4::UVec4(const JPH::UVec4&)’
   27 |                                                                 UVec4(const UVec4 &inRHS) = default;
      |                                                                 ^~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:27:65: note:   candidate expects 1 argument, 4 provided
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:26:65: note: candidate: ‘JPH::UVec4::UVec4()’
   26 |                                                                 UVec4() = default; ///< Intentionally not initialized for performance reasons
      |                                                                 ^~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/UVec4.h:26:65: note:   candidate expects 0 arguments, 4 provided
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl: In static member function ‘static JPH::Mat44 JPH::Mat44::sQuatLeftMultiply(JPH::QuatArg)’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl:1106:100: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_W, JPH::SWIZZLE_Z, JPH::SWIZZLE_Y, JPH::SWIZZLE_X>()’
 1106 |                 Vec4(1, 1, -1, -1) * inQ.mValue.Swizzle<SWIZZLE_W, SWIZZLE_Z, SWIZZLE_Y, SWIZZLE_X>(),
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl:1107:100: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_W, JPH::SWIZZLE_X, JPH::SWIZZLE_Y>()’
 1107 |                 Vec4(-1, 1, 1, -1) * inQ.mValue.Swizzle<SWIZZLE_Z, SWIZZLE_W, SWIZZLE_X, SWIZZLE_Y>(),
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl:1108:100: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_X, JPH::SWIZZLE_W, JPH::SWIZZLE_Z>()’
 1108 |                 Vec4(1, -1, 1, -1) * inQ.mValue.Swizzle<SWIZZLE_Y, SWIZZLE_X, SWIZZLE_W, SWIZZLE_Z>(),
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl: In static member function ‘static JPH::Mat44 JPH::Mat44::sQuatRightMultiply(JPH::QuatArg)’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl:1115:100: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_W, JPH::SWIZZLE_Z, JPH::SWIZZLE_Y, JPH::SWIZZLE_X>()’
 1115 |                 Vec4(1, -1, 1, -1) * inQ.mValue.Swizzle<SWIZZLE_W, SWIZZLE_Z, SWIZZLE_Y, SWIZZLE_X>(),
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl:1116:100: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Z, JPH::SWIZZLE_W, JPH::SWIZZLE_X, JPH::SWIZZLE_Y>()’
 1116 |                 Vec4(1, 1, -1, -1) * inQ.mValue.Swizzle<SWIZZLE_Z, SWIZZLE_W, SWIZZLE_X, SWIZZLE_Y>(),
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/Mat44.inl:1117:100: error: no matching function for call to ‘JPH::Vec4::Swizzle<JPH::SWIZZLE_Y, JPH::SWIZZLE_X, JPH::SWIZZLE_W, JPH::SWIZZLE_Z>()’
 1117 |                 Vec4(-1, 1, 1, -1) * inQ.mValue.Swizzle<SWIZZLE_Y, SWIZZLE_X, SWIZZLE_W, SWIZZLE_Z>(),
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note: candidate: ‘template<<declaration error>, <declaration error>, <declaration error>, <declaration error> > JPH::Vec4 JPH::Vec4::Swizzle() const’
  182 |         JPH_INLINE Vec4                         Swizzle() const;
      |                                                 ^~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Vec4.h:182:49: note:   template argument deduction/substitution failed:
In file included from /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/Real.h:7,
                 from /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/..//Jolt/Jolt.h:16:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/DVec3.h: At global scope:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/DVec3.h:270:66: error: ‘uint64’ was not declared in this scope; did you mean ‘uint’?
  270 |         inline static const double      cTrue = BitCast<double>(~uint64(0));
      |                                                                  ^~~~~~
      |                                                                  uint
In file included from /home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Build/../Jolt/Math/DVec3.h:284:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/DVec3.inl: In member function ‘JPH::DVec3 JPH::DVec3::PrepareRoundToZero() const’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/DVec3.inl:833:19: error: ‘uint64’ does not name a type; did you mean ‘uint’?
  833 |         constexpr uint64 cDoubleToFloatMantissaLoss = (1U << 29) - 1;
      |                   ^~~~~~
      |                   uint
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/DVec3.inl:836:86: error: ‘cDoubleToFloatMantissaLoss’ was not declared in this scope
  836 |         return _mm256_and_pd(mValue, _mm256_castsi256_pd(_mm256_set1_epi64x(int64_t(~cDoubleToFloatMantissaLoss))));
      |                                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/DVec3.inl: In member function ‘JPH::DVec3 JPH::DVec3::PrepareRoundToInf() const’:
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/DVec3.inl:855:19: error: ‘uint64’ does not name a type; did you mean ‘uint’?
  855 |         constexpr uint64 cDoubleToFloatMantissaLoss = (1U << 29) - 1;
      |                   ^~~~~~
      |                   uint
/home/dennis/src/vcpkg/buildtrees/joltphysics/src/v3.0.1-aeee633c24.clean/Jolt/Math/DVec3.inl:858:52: error: ‘cDoubleToFloatMantissaLoss’ was not declared in this scope
  858 |         __m256i mantissa_loss = _mm256_set1_epi64x(cDoubleToFloatMantissaLoss);
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```
</details>**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "dependencies": [
    "joltphysics"
  ]
}

```
</details>

--- a/ports/joltphysics/fix-export.diff
+++ b/ports/joltphysics/fix-export.diff
@@ -1,8 +1,8 @@
 diff --git a/Build/CMakeLists.txt b/Build/CMakeLists.txt
-index e4fddc50..655cdc80 100644
+index cc4cf1d..330b987 100644
 --- a/Build/CMakeLists.txt
 +++ b/Build/CMakeLists.txt
-@@ -91,8 +91,8 @@ if (("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows" OR "${CMAKE_SYSTEM_NAME}" STREQUA
+@@ -100,8 +100,8 @@ if (("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows" OR "${CMAKE_SYSTEM_NAME}" STREQUA
  	endif()
  
  	# Set compiler flags for various configurations
@@ -13,7 +13,7 @@ index e4fddc50..655cdc80 100644
  	set(CMAKE_CXX_FLAGS_DISTRIBUTION "/GS- /Gy /O2 /Oi /Ot")
  	set(CMAKE_CXX_FLAGS_RELEASEASAN "-fsanitize=address /Od")
  	set(CMAKE_CXX_FLAGS_RELEASEUBSAN "-fsanitize=undefined,implicit-conversion,float-divide-by-zero,local-bounds -fno-sanitize-recover=all")
-@@ -148,8 +148,8 @@ elseif ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR "${CMAKE_SYSTEM_NAME}" STREQU
+@@ -160,8 +160,8 @@ elseif ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR "${CMAKE_SYSTEM_NAME}" STREQU
  	endif()
  
  	# Set compiler flags for various configurations
@@ -24,7 +24,7 @@ index e4fddc50..655cdc80 100644
  	set(CMAKE_CXX_FLAGS_DISTRIBUTION "-O3")
  	set(CMAKE_CXX_FLAGS_RELEASEASAN "-fsanitize=address")
  	set(CMAKE_CXX_FLAGS_RELEASEUBSAN "-fsanitize=undefined,implicit-conversion,float-divide-by-zero,local-bounds -fno-sanitize-recover=all")
-@@ -193,7 +193,11 @@ if (IOS)
+@@ -205,7 +205,11 @@ if (XCODE)
  endif()
  
  # Install Jolt library and includes
@@ -37,7 +37,7 @@ index e4fddc50..655cdc80 100644
  foreach(SRC_FILE ${JOLT_PHYSICS_SRC_FILES})
  	string(REPLACE ${PHYSICS_REPO_ROOT} "" RELATIVE_SRC_FILE ${SRC_FILE})
  	get_filename_component(DESTINATION_PATH ${RELATIVE_SRC_FILE} DIRECTORY)
-@@ -202,6 +206,17 @@ foreach(SRC_FILE ${JOLT_PHYSICS_SRC_FILES})
+@@ -214,6 +218,17 @@ foreach(SRC_FILE ${JOLT_PHYSICS_SRC_FILES})
  	endif()
  endforeach()
  
@@ -52,17 +52,17 @@ index e4fddc50..655cdc80 100644
 +  DESTINATION share/unofficial-joltphysics
 +)
 +
- # Check if we're the root CMakeLists.txt, if not we are included by another CMake file and we should disable everything except for the main library 
+ # Check if we're the root CMakeLists.txt, if not we are included by another CMake file and we should disable everything except for the main library
  if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
  	# Ability to turn ON/OFF individual applications
 diff --git a/Jolt/Jolt.cmake b/Jolt/Jolt.cmake
-index 176a7578..b31b2417 100644
+index c0f7099..d447a3f 100644
 --- a/Jolt/Jolt.cmake
 +++ b/Jolt/Jolt.cmake
-@@ -431,10 +431,13 @@ source_group(TREE ${JOLT_PHYSICS_ROOT} FILES ${JOLT_PHYSICS_SRC_FILES})
+@@ -474,10 +474,13 @@ if (BUILD_SHARED_LIBS)
+ 	target_compile_definitions(Jolt PRIVATE JPH_BUILD_SHARED_LIBRARY)
+ endif()
  
- # Create Jolt lib
- add_library(Jolt STATIC ${JOLT_PHYSICS_SRC_FILES})
 -target_include_directories(Jolt PUBLIC ${PHYSICS_REPO_ROOT})
 +target_include_directories(Jolt PUBLIC
 +  $<BUILD_INTERFACE:${PHYSICS_REPO_ROOT}>

--- a/ports/joltphysics/portfile.cmake
+++ b/ports/joltphysics/portfile.cmake
@@ -3,8 +3,8 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO jrouwe/JoltPhysics
-    REF "v${VERSION}"
-    SHA512 367e5b945e8f91a0c0c9eb699db6f49351aa39b0af9b8fd0be5f474d65b28a7244880eedad10cbd2db0e031daa28bbabb5f9fb8bf9af653dd1f86904bfde44a2
+    REF 4e038d190b7e97df01e6a81c8f74551fe487a09d # Commited on 2023-11-04
+    SHA512 d80383ac0dfadda6ff353bd241c3af9ab879de4f6cba04db7dc0e82ede2a4f120b1af9baca63000cefb06778447105bfd6bd26a9a6b8fa36fc5daaa75772bdf2
     HEAD_REF master
     PATCHES
       fix-export.diff

--- a/ports/joltphysics/portfile.cmake
+++ b/ports/joltphysics/portfile.cmake
@@ -3,8 +3,8 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO jrouwe/JoltPhysics
-    REF 4e038d190b7e97df01e6a81c8f74551fe487a09d # Commited on 2023-11-04
-    SHA512 d80383ac0dfadda6ff353bd241c3af9ab879de4f6cba04db7dc0e82ede2a4f120b1af9baca63000cefb06778447105bfd6bd26a9a6b8fa36fc5daaa75772bdf2
+    REF "v${VERSION}"
+    SHA512 9b7530c37fc865682c4a130afc87daef1b038d1f457d2330a5253f35e3a4b6399ad738e97961f0ca8a9ae41ed999179e1c864dc699c5c93341ce4b6e6b2a1a61
     HEAD_REF master
     PATCHES
       fix-export.diff

--- a/ports/joltphysics/vcpkg.json
+++ b/ports/joltphysics/vcpkg.json
@@ -1,7 +1,6 @@
 {
   "name": "joltphysics",
-  "version": "3.0.1",
-  "port-version": 1,
+  "version-date": "2023-11-04",
   "description": "A multi core friendly rigid body physics and collision detection library suitable for games and VR applications",
   "homepage": "https://github.com/jrouwe/JoltPhysics",
   "license": "MIT",

--- a/ports/joltphysics/vcpkg.json
+++ b/ports/joltphysics/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "joltphysics",
-  "version-date": "2023-11-04",
+  "version": "4.0.0",
   "description": "A multi core friendly rigid body physics and collision detection library suitable for games and VR applications",
   "homepage": "https://github.com/jrouwe/JoltPhysics",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3617,7 +3617,7 @@
       "port-version": 3
     },
     "joltphysics": {
-      "baseline": "2023-11-04",
+      "baseline": "4.0.0",
       "port-version": 0
     },
     "josuttis-jthread": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3617,8 +3617,8 @@
       "port-version": 3
     },
     "joltphysics": {
-      "baseline": "3.0.1",
-      "port-version": 1
+      "baseline": "2023-11-04",
+      "port-version": 0
     },
     "josuttis-jthread": {
       "baseline": "2020-07-21",

--- a/versions/j-/joltphysics.json
+++ b/versions/j-/joltphysics.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "8077d0a7c05b6a373d765af5a7e759746df37993",
+      "version-date": "2023-11-04",
+      "port-version": 0
+    },
+    {
       "git-tree": "f8fca568d1ae240c8d3ca23ae3b9a09f6fb518ba",
       "version": "3.0.1",
       "port-version": 1

--- a/versions/j-/joltphysics.json
+++ b/versions/j-/joltphysics.json
@@ -6,11 +6,6 @@
       "port-version": 0
     },
     {
-      "git-tree": "8077d0a7c05b6a373d765af5a7e759746df37993",
-      "version-date": "2023-11-04",
-      "port-version": 0
-    },
-    {
       "git-tree": "f8fca568d1ae240c8d3ca23ae3b9a09f6fb518ba",
       "version": "3.0.1",
       "port-version": 1

--- a/versions/j-/joltphysics.json
+++ b/versions/j-/joltphysics.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f7ea4a51e3f8cfa76e079f80d8ece87ef6695553",
+      "version": "4.0.0",
+      "port-version": 0
+    },
+    {
       "git-tree": "8077d0a7c05b6a373d765af5a7e759746df37993",
       "version-date": "2023-11-04",
       "port-version": 0

